modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -142,7 +142,10 @@ def merge(self, right, **kwargs):
         left_index = kwargs.get("left_index", False)
         right_index = kwargs.get("right_index", False)
         """Only non-index joins with explicit 'on' are supported"""
-        if left_index is False and right_index is False and on is not None:
+        if left_index is False and right_index is False:
+            if on is None:
+                on = [c for c in self.columns if c in right.columns]
+
```

Comment:
I see this condition below in the code
```python
if not isinstance(on, list):
    assert isinstance(on, str), f"unsupported 'on' value {on}"
    on = [on]
```
Doesn't our OmniSci backend handle this case?
```python
import modin.pandas as pd
df = pd.DataFrame([[1,2],[3,4]])
df2 = pd.DataFrame([[1,2],[3,4]])
df.merge(df2, on=1)
   0_x  1  0_y
0    1  2    1
1    3  4    3
```

Reply:
Initially, we didn't work for non-string column names in the engine. Currently, this condition can be probably relaxed. I'll check it in a separate patch.

Reply:
Great, thanks!